karma/module.py

```diff
@@ -0,0 +1,641 @@
+                    emoji_str = f"<:pumpkin:{emoji.emoji_id}>"
+
+                idx = i // 8
+                if i % 8 == 0:
+                    emoji_lists[idx] = []
+                emoji_lists[idx].append(emoji_str)
+
+            lines = [" ".join(line) for line in emoji_lists.values()]
+            return lines
+
+        if len(emojis_positive):
+            await ctx.send(_(ctx, "Emojis with positive karma"))
+            for line in format_emojis(emojis_positive):
+                if line:
+                    await ctx.send(line)
+        if len(emojis_neutral):
+            await ctx.send(_(ctx, "Emojis with neutral karma"))
+            for line in format_emojis(emojis_neutral):
+                if line:
+                    await ctx.send(line)
+        if len(emojis_negative):
+            await ctx.send(_(ctx, "Emojis with negative karma"))
+            for line in format_emojis(emojis_negative):
+                if line:
+                    await ctx.send(line)
+
+    @commands.check(check.acl)
+    @karma_.command(name="vote")
+    async def karma_vote(self, ctx, emoji: str = None):
+        """Vote over emoji's karma value."""
+        await utils.Discord.delete_message(ctx.message)
+
+        with contextlib.suppress(commands.EmojiNotFound):
+            emoji = await commands.EmojiConverter().convert(ctx, emoji)
+
+        if emoji is None:
+            voted_ids = [e.emoji_id for e in DiscordEmoji.get_all(ctx.guild.id)]
+            for guild_emoji in ctx.guild.emojis:
+                if guild_emoji.id not in voted_ids:
+                    emoji = guild_emoji
+                    break
+
+        if emoji is None:
+            await ctx.author.send(
+                _(ctx, "All server emojis have been assigned a karma value.")
+            )
+            return
+        emoji_name: str = getattr(emoji, "name", str(emoji))
+
+        message = (
+            _(ctx, "Karma vote over the value of {emoji} started.")
+            + "\n"
+            + _(ctx, "The vote will run for **{minutes}** minutes.")
+            + " "
+            + _(ctx, "Required minimum vote count is **{count}**.")
+        )
+        # TODO Make configurable
+        vote_message = await ctx.send(
+            message.format(emoji=str(emoji), minutes=120, count=10)
+        )
+
+        # Set the value to zero, so we can run this command multiple times
+        # without starting a vote over the same emoji over and over.
+        if type(emoji) is discord.Emoji:
+            DiscordEmoji.add(ctx.guild.id, emoji.id, 0)
+
+        await guild_log.info(
+            ctx.author, ctx.channel, f"Karma vote over emoji '{emoji_name}' started."
+        )
+
+        votes = {"🔼": 0, "0⃣": 0, "🔽": 0}
+        emoji_labels = {"🔼": "+1", "0⃣": "0", "🔽": "-1"}
+        for vote_option in votes.keys():
+            await vote_message.add_reaction(vote_option)
+
+        # TODO Make configurable
+        await asyncio.sleep(10)
```

Comment:
What is the easiest way to make it configurable? How to save the preferences?

Reply:
if you really want to make this configurable, the only way I see is with a setting in the database that could be set by a command